python-avd/pyavd/_eos_designs/structured_config/network_services/ethernet_interfaces.py

```diff
@@ -22,265 +25,229 @@ class EthernetInterfacesMixin(Protocol):
-
-                            if l3_interface.pim.enabled:
-                                if not getattr(vrf._internal_data, "evpn_l3_multicast_enabled", False):
-                                    # Possibly the key was not set because `evpn_multicast` is not set to `true`.
-                                    if not self.shared_utils.evpn_multicast:
-                                        msg = (
-                                            f"'pim: enabled' set on l3_interface '{interface_name}' on '{self.shared_utils.hostname}' requires "
-                                            "'evpn_multicast: true' at the fabric level"
-                                        )
-                                    else:
-                                        msg = (
-                                            f"'pim: enabled' set on l3_interface '{interface_name}' on '{self.shared_utils.hostname}' requires "
-                                            f"'evpn_l3_multicast.enabled: true' under VRF '{vrf.name}' or Tenant '{tenant.name}'"
-                                        )
-                                    raise AristaAvdError(msg)
-
-                                if not getattr(vrf._internal_data, "pim_rp_addresses", None):
-                                    msg = (
-                                        f"'pim: enabled' set on l3_interface '{interface_name}' on '{self.shared_utils.hostname}' requires at least one RP"
-                                        f" defined in pim_rp_addresses under VRF '{vrf.name}' or Tenant '{tenant.name}'"
-                                    )
-                                    raise AristaAvdError(msg)
-
-                                interface["pim"] = {"ipv4": {"sparse_mode": True}}
-
-                            # Strip None values from vlan before adding to list
-                            interface = {key: value for key, value in interface.items() if value is not None}
-
-                            append_if_not_duplicate(
-                                list_of_dicts=ethernet_interfaces,
-                                primary_key="name",
-                                new_dict=interface,
-                                context="Ethernet Interfaces defined under l3_interfaces",
-                                context_keys=["name", "vrf"],
-                            )
+                    self._set_l3_interfaces(vrf, tenant, subif_parent_interface_names)
 
         if self.shared_utils.network_services_l1:
             for tenant in self.shared_utils.filtered_tenants:
                 if not tenant.point_to_point_services:
                     continue
+                self._set_point_to_point_interfaces(tenant, subif_parent_interface_names)
+
+        # Add missing parent interface names if any
+        if missing_parent_interface_names := subif_parent_interface_names.difference(eth_int.name for eth_int in self.structured_config.ethernet_interfaces):
```

Comment:
Once we flip the global toggle to not create a structured config per module, this will inspect _all_ ethernet_interfaces created up until this point. Maybe we should use .obtain and check that the interface has no switchport and error out otherwise. Can be done later ofc.

Reply:
Added an issue to track this though it does mean that this becomes order sensitive. Could this become a problem?